packages/api/endpoints/granules.js

```diff
@@ -1,144 +1,221 @@
 'use strict';
 
-const _get = require('lodash.get');
 const aws = require('@cumulus/common/aws');
-const { inTestMode } = require('@cumulus/common/test-utils');
 const log = require('@cumulus/common/log');
-const handle = require('../lib/response').handle;
+const {
+  buildLambdaProxyResponse,
+  getAuthorizationFailureResponse
+} = require('../lib/response');
 const Search = require('../es/search').Search;
 const models = require('../models');
 
 /**
  * List all granules for a given collection.
  *
  * @param {Object} event - aws lambda event object.
- * @param {callback} cb - aws lambda callback function
- * @returns {Object} list of granules
+ * @returns {Promise<Object>} a Lambda Proxy response object
  */
-function list(event, cb) {
-  const search = new Search(event, 'granule');
-  return search.query().then((res) => cb(null, res)).catch(cb);
+async function list(event) {
+  const result = await (new Search(event, 'granule')).query();
+
+  return buildLambdaProxyResponse({
+    json: true,
+    body: result
+  });
 }
 
 /**
  * Update a single granule.
  * Supported Actions: reingest, applyWorkflow, RemoveFromCMR.
  *
  * @param {Object} event - aws lambda event object.
- * @returns {Promise} response from the actions
+ * @returns {Promise<Object>} a Lambda Proxy response object
  */
 async function put(event) {
-  const granuleId = _get(event.pathParameters, 'granuleName');
-  let body = _get(event, 'body', '{}');
-  body = JSON.parse(body);
+  const granuleId = event.pathParameters.granuleName;
+  const body = event.body ? JSON.parse(event.body) : {};
+  const action = body.action;
+
+  if (!action) {
+    return buildLambdaProxyResponse({
+      json: true,
+      statusCode: 400,
+      body: { message: 'Action is missing' }
+    });
+  }
 
-  const action = _get(body, 'action');
-  const g = new models.Granule();
+  const granuleModelClient = new models.Granule();
+  const granule = await granuleModelClient.get({ granuleId });
 
-  if (action) {
-    const response = await g.get({ granuleId });
-    if (action === 'reingest') {
-      return await g.reingest(response);
-    }
-    if (action === 'applyWorkflow') {
-      const workflow = _get(body, 'workflow');
-      const messageSource = _get(body, 'messageSource');
-      const metaOverride = _get(body, 'metaOverride');
-      const payloadOverride = _get(body, 'payloadOverride');
-      return await g.applyWorkflow(response, workflow, messageSource, metaOverride, payloadOverride);
-    }
-    else if (action === 'removeFromCmr') {
-      await g.removeGranuleFromCmr(response.granuleId, response.collectionId);
-      return {
-        granuleId: response.granuleId,
+  if (action === 'reingest') {
+    await granuleModelClient.reingest(granule);
+
+    return buildLambdaProxyResponse({
+      json: true,
+      body: {
```

Comment:
since granuleModelClient.reingest returns this same object, can we avoid hard coding it into the body here?

Reply:
We can also remove that functionality from the granules model. The point is that the API response should only be built in one place.

Reply:
Agreed, I'll make those changes.